app/jobs/cache/record_counts/data_providers_job.rb

```diff
@@ -7,9 +7,24 @@ class DataProvidersJob < ActiveJob::Base
 
       queue_as :default
 
-      def perform(provider)
-        params = { query: '*:*', rows: 0, profile: 'minimal facets', facet: 'DATA_PROVIDER', qf: "PROVIDER:\"#{provider}\"" }
-        response = repository.search(params)
+      def perform(provider, collection_id = nil)
```

Comment:
Assignment Branch Condition size for perform is too high. [25.18/15]<br>Method has too many lines. [21/10]